File: 0704_binary_search/binary_search.py
from typing import List


def search(nums: List[int], target: int) -> int:
    """
    Time: O(log(n))
    Space: (1)
    """
    i, j = 0, len(nums)
    while i < j:
        mid = (i + j) // 2
        val = nums[mid]
        if val == target:
            return mid
        if target < val:
            j = mid
        else:
            i = mid + 1
    return -1

File: 0704_binary_search/binary_search.java
class Solution {
    // Time: O(log(n))
    // Space: (1)
    public int search(int[] nums, int target) {
        int i = 0, j = nums.length - 1;
        while (i <= j) {
            int mid = (i + j) / 2;
            int val = nums[mid];
            if (val == target) {
                return mid;
            }
            if (target < val) {
                j = mid - 1;
            } else {
                i = mid + 1;
            }
        }
        return -1;
    }
}